core/src/main/java/org/apache/iceberg/ManifestWriter.java

```diff
@@ -31,9 +31,50 @@
 /**
  * Writer for manifest files.
  */
-class ManifestWriter implements FileAppender<DataFile> {
+public class ManifestWriter implements FileAppender<DataFile> {
   private static final Logger LOG = LoggerFactory.getLogger(ManifestWriter.class);
 
+  static ManifestFile copyAppendManifest(ManifestReader reader, OutputFile outputFile, long snapshotId,
+                                         SnapshotSummary.Builder summaryBuilder) {
+    ManifestWriter writer = new ManifestWriter(reader.spec(), outputFile, snapshotId);
+    boolean threw = true;
+    try {
+      for (ManifestEntry entry : reader.entries()) {
+        Preconditions.checkArgument(entry.status() == ManifestEntry.Status.ADDED,
+            "Cannot append manifest: contains existing files");
+        summaryBuilder.addedFile(reader.spec(), entry.file());
```

Comment:
This particular feature would come very handy for use-cases that involve manifests migration or manifest sanitization even and I was wondering if we could consider a very basic extension to the API, something that would enable us to also mangle with the `DataFile` details?
I guess a very basic `Function<DataFile, DataFile> fileHandler` to wrap the `entry.file()` or something similar would do the trick; also it'd be very simple to provide the identity function for the default implementation.
Wdyt?

Reply:
I'm not sure that I follow what you're suggesting. Could you write it up and propose it on the dev list?

Reply:
Will do